example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,565 @@
+        [0, Number.MIN_VALUE],
+        [-10000, 0],
+        Extrapolation.CLAMP
+      );
+      showRightProgress.value =
+        rightWidth.value > 0
+          ? interpolate(
+              appliedTranslation.value,
+              [-rightWidth.value, 0, 1],
+              [1, 0, 0]
+            )
+          : 0;
+      rightActionTranslate.value = interpolate(
+        showRightProgress.value,
+        [0, Number.MIN_VALUE],
+        [-10000, 0],
+        Extrapolation.CLAMP
+      );
+    };
+
+    const animateRow = (
+      fromValue: number,
+      toValue: number,
+      velocityX?: number
+    ) => {
+      'worklet';
+
+      rowState.value = Math.sign(toValue);
+
+      appliedTranslation.value = withSpring(
+        toValue,
+        {
+          duration: 1000,
+          dampingRatio: 1.2,
+          stiffness: 500,
+          velocity: velocityX,
+          ...props.animationOptions,
+        },
+        (isFinished) => {
+          if (isFinished) {
+            if (toValue > 0 && props.onSwipeableOpen) {
+              runOnJS(props.onSwipeableOpen)('left', swipeableMethods);
+            } else if (toValue < 0 && props.onSwipeableOpen) {
+              runOnJS(props.onSwipeableOpen)('right', swipeableMethods);
+            } else if (props.onSwipeableClose) {
+              const closingDirection = fromValue > 0 ? 'left' : 'right';
+              runOnJS(props.onSwipeableClose)(
+                closingDirection,
+                swipeableMethods
+              );
+            }
+          }
+        }
+      );
+
+      if (toValue > 0 && props.onSwipeableWillOpen) {
+        runOnJS(props.onSwipeableWillOpen)('left');
+      } else if (toValue < 0 && props.onSwipeableWillOpen) {
+        runOnJS(props.onSwipeableWillOpen)('right');
+      } else if (props.onSwipeableWillClose) {
+        const closingDirection = fromValue > 0 ? 'left' : 'right';
+        runOnJS(props.onSwipeableWillClose)(closingDirection);
+      }
+    };
+
+    const onRowLayout = ({ nativeEvent }: LayoutChangeEvent) => {
+      rowWidth.value = nativeEvent.layout.width;
+    };
+
+    const {
+      children,
+      renderLeftActions,
+      renderRightActions,
+      dragOffsetFromLeftEdge = 10,
+      dragOffsetFromRightEdge = 10,
+    } = props;
+
+    swipeableMethods = useMemo<SwipeableMethods>(
+      () => ({
+        close() {
+          'worklet';
+          animateRow(calculateCurrentOffset(), 0);
+        },
+        openLeft() {
+          'worklet';
+          animateRow(calculateCurrentOffset(), leftWidth.value);
+        },
+        openRight() {
+          'worklet';
+          rightWidth.value = rowWidth.value - rightOffset.value;
+          animateRow(calculateCurrentOffset(), -rightWidth.value);
+        },
+        reset() {
+          'worklet';
+          userDrag.value = 0;
+          appliedTranslation.value = 0;
+          rowState.value = 0;
+        },
+      }),
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is this necessary when you have dependencies below?

Reply:
not all are there, adding all of them would result in updating the useMemo every frame